run_service.sh

```diff
@@ -880,15 +886,26 @@ remote_service_hash=$(poetry run python "../scripts/service_hash.py")
 operator_address=$(get_address "../$operator_keys_file")
 on_chain_agent_id=$(get_on_chain_agent_ids "$service_id")
 
-if [ "$local_service_hash" != "$remote_service_hash" ] || [ "$on_chain_agent_id" != "$AGENT_ID" ]; then
+# On-chain agent bond for the expected agent ID ($AGENT_ID)
+on_chain_agent_bond=$(poetry run python "../scripts/get_agent_bond.py" "$CUSTOM_SERVICE_REGISTRY_TOKEN_UTILITY_ADDRESS" "$service_id" "$AGENT_ID" "$rpc")
+
+if [ "${USE_STAKING}" = true ]; then
+    cost_of_bonding=$MIN_STAKING_BOND_OLAS
+else
+    cost_of_bonding=$MIN_STAKING_BOND_XDAI
+fi
+
+if [ "$local_service_hash" != "$remote_service_hash" ] || [ "$on_chain_agent_id" != "$AGENT_ID" ] || [ "$on_chain_agent_bond" != "$cost_of_bonding" ]; then
```

Comment:
If hash, agent_id or bond don't match the expected values, it requires an on-chain service update.